app/util/process_utils.py

```diff
@@ -76,6 +76,9 @@ def Popen_with_delayed_expansion(cmd, *args, **kwargs):
         else:
             cmd_with_deplayed_expansion.extend(cmd)
         cmd = cmd_with_deplayed_expansion
+    else:
+        if kwargs.get('shell') == True:
+            cmd = 'set -o pipefail; ' + cmd
```

Comment:
Can you add a short comment to explain why this is needed since it might not be that obvious.